src/main/java/org/javarosa/core/model/SelectChoice.java

```diff
@@ -91,15 +111,33 @@ public String getValue() {
     }
 
     public String getChild(String childName) {
-        if (item != null) {
-            TreeElement child = item.getChild(childName, 0);
-            if (child != null) {
-                return child.getValue().getDisplayText();
-            }
+        if (item == null) {
```

Comment:
One thing I find hard about JR is that it can be unclear what's a deliberate precondition vs a case that wasn't explicitly considered vs ...? I think this precondition is now much clearer even though it's equivalent to the terser code it replaces. (As I mentioned, this thinking about preconditions is what had nudged me in the direction of runtime exceptions.)